src/components/PeopleManagement/EditGroupNameModal.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ActionRow, Button, Form, ModalDialog,
+} from '@openedx/paragon';
+import { logError } from '@edx/frontend-platform/logging';
+
+import LmsApiService from '../../data/services/LmsApiService';
+
+const EditGroupNameModal = ({
+  group, isOpen, close, setShowToast, setToastMessage,
+}) => {
+  const [name, setName] = useState(group?.name);
+  const editEnterpriseGroup = async () => {
+    try {
+      const formData = { name };
+      const response = await LmsApiService.updateEnterpriseGroup(group?.uuid, formData);
+      if (response.status === 200) {
+        setShowToast(true);
+        setToastMessage('Group name updated');
+      }
+    } catch (error) {
+      logError(error);
+    }
+  };
+
+  return (
+    <ModalDialog
+      title="Edit group"
+      isOpen={isOpen}
+      onClose={close}
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>Edit group name</ModalDialog.Title>
+      </ModalDialog.Header>
+
+      <ModalDialog.Body>
+        <Form.Group>
+          <Form.Control
+            floatingLabel="Group name"
+            value={name}
+            data-testid="group name input"
+            onChange={(e) => setName(e.target.value)}
+          />
+        </Form.Group>
+      </ModalDialog.Body>
+
+      <ModalDialog.Footer>
+        <ActionRow>
+          <ModalDialog.CloseButton variant="tertiary">
+            Cancel
+          </ModalDialog.CloseButton>
+          <Button variant="primary" onClick={editEnterpriseGroup}>Done</Button>
```

Comment:
let's consider making this a stateful button in case of an error or a slow request, we'd want to show some sort of feedback to the user.